docs/changelog/assets/js/app.js

```diff
@@ -0,0 +1,442 @@
+        disable.version = true;
+    }
+    else{
+        disable.searchButton = false;
+    }
+    
+    if(formParams.version && formParams.version.trim() !== ''){
+        disable.version = false;
+        disable.commitMessage = true;
+        disable.commitHash = true;
+        disable.searchButton = false;
+    }
+    else if((formParams.commitMessage && formParams.commitMessage.trim() !== '') || (formParams.commitHash && formParams.commitHash.trim() !== '')){
+        disable.version = true;
+        disable.searchButton = false;
+    }
+
+    for(let key in disable){
+        switch(key){
+            case 'package':
+                if(disable[key]){
+                    packageNameInputDropdown.disabled = true;
+                    packageNameInputDropdown.value = "";
+                    packageInputGroup.classList.add('hide');
+                    formParams.package = null;
+                }
+                else{
+                    packageNameInputDropdown.disabled = false;
+                    packageInputGroup.classList.remove('hide');
+                }
+                break;
+            case 'version':
+                if(disable[key]){
+                    versionInput.disabled = true;
+                    versionInput.value = "";
+                    versionInputGroup.classList.add('hide');
+                    formParams.version = null;
+                }
+                else{
+                    versionInput.disabled = false;
+                    versionInputGroup.classList.remove('hide');
+                }
+                break;
+            case 'commitMessage':
+                if(disable[key]){
+                    commitMessageInput.disabled = true;
+                    commitMessageInput.value = "";
+                    commitMessageGroup.classList.add('hide');
+                    formParams.commitMessage = null;
+                }
+                else{
+                    commitMessageInput.disabled = false;
+                    commitMessageGroup.classList.remove('hide');
+                }
+                break;
+            case 'commitHash':
+                if(disable[key]){
+                    commitHashInput.disabled = true;
+                    commitHashInput.value = "";
+                    commitHashGroup.classList.add('hide');
+                    formParams.commitHash = null;
+                }
+                else{
+                    commitHashInput.disabled = false;
+                    commitHashGroup.classList.remove('hide');
+                }
+                break;
+            case 'searchButton':
+                searchButton.disabled = disable[key];
+                break;
+        }
+    }
+};
+
+const doSearch_commit = (searchParams, drill_down) => {
+    let resulting_versions = new Set(),
+        resulting_commit_messages = new Set(),
+        resulting_commit_hash = new Set(),
+        search_results = [];
+    for(let package in drill_down){
```

Comment:
This function would eventually get really slow as the changelogs increase. I don't have a solution either maybe we can discuss this with the team.

Reply:
I agree too. However, this is once in a while search. 

I am sure, people are going to mostly look for commit hash or package version. Plus, whenever there's a stable release, we are going to switch the JSON and this search is not across the JSONs. So, it's okay for now.